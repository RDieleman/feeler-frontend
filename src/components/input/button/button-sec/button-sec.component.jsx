import React from "react";
import "./button-sec.styles.css";
import "../button.styles.css";

export const ButtonSecComponent = ({content, iconLocation, handleOnClick}) =>{
    // Secondary button
    return(
        <button className="text-body-sec button-sec" onClick={() => handleOnClick()}>
            {content}
            {/*Add icon if location is included*/}
            {(iconLocation === undefined)? "" :
                // todo: Replace with svg and set color in css file
                <img src={iconLocation} alt="btn-icon"/>}
        </button>
    )
}