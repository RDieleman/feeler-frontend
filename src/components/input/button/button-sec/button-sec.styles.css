.button-sec{
    background-color: var(--color-gen-light);
    border: var(--border-width) solid var(--color-main-medium);
    border-radius: var(--border-radius);

    text-transform: none;

    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    color: var(--color-main-medium);
}

.button-sec > div:first-child{
    flex: 1;
}

.button-sec > img{
    height: 100%;
    width: auto;;
}