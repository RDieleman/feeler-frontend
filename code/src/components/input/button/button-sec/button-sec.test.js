import React from "react";
import {shallow} from "enzyme";
import {ButtonSecComponent} from "./button-sec.component";

it("Expect to render ButtonSec component", () => {
    expect(shallow(
        <ButtonSecComponent
            handleOnClick={() => "ButtonSec Clicked"}
            content="test"
            iconLocation="test"
        />)).toMatchSnapshot();
});