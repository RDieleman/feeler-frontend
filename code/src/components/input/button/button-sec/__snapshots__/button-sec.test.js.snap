// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render ButtonSec component 1`] = `ShallowWrapper {}`;
